### PR TITLE
Exclude WinMD AttributeNameAttribute

### LINKED ISSUE #8430: WindowsTargetPlatformVersion Bump Breaks Existing Community Modules
In https://github.com/microsoft/react-native-windows/pull/7779 (included in 0.66) we bumped the target Windows platform version to the default brought in by VS. Existing community modules like react-native-picker will still use an older version of it https://github.com/react-native-picker/picker/blob/2eda2f7216ec7ace345f36c1483e0881f8132755/windows/ReactNativePicker/ReactNativePicker.vcxproj#L16

Community module MIDL will include references to MSRN MIDL, and there seems to be a conflict when the SDK versions do not line up.

image

```
     8>midlrt : error MIDL1002: [msg]error while reading input file [context]Error processing WinMD metadata. Custom at
       tribute Microsoft.ReactNative.DocDefaultAttribute: Custom attribute Windows.Foundation.Metadata.AttributeNameAtt
       ribute: Custom attribute Windows.Foundation.Metadata.AttributeNameAttribute could not be found. [D:\react-native
       -windows\node_modules\@react-native-picker\picker\windows\ReactNativePicker\ReactNativePicker.vcxproj]
     8>D:\react-native-windows\node_modules\@react-native-picker\picker\windows\ReactNativePicker\ReactPickerView.idl(1
       ): error MIDL1002: [msg]error while reading input file [D:\react-native-windows\node_modules\@react-native-picke
       r\picker\windows\ReactNativePicker\ReactNativePicker.vcxproj]
```

Locally modifying the `WindowsTargetPlatformVersion` in the community module seems to resolve the issue, but this would be a break to the entire ecosystem.

### PR DESCRIPTION
**Pull request by NickGerleman**
Fixes #8430

Newer versions of the Windows SDK by default produce WinMDs that may not be able to be referenced when using an older Windows SDK. This change disables the new feature to allow existing community modules targeting an older Windows SDK to consume the MSRN WinMDs.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8439)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj [hunk 1]
```diff
@@ -146,6 +146,8 @@
     </Link>
     <Midl>
       <AdditionalIncludeDirectories>$(ReactNativeWindowsDir)Microsoft.ReactNative\Views\cppwinrt;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <!-- #8430: This is needed to allow modules targeting older SDKs to consume the geneerated WinMD -->
+      <AdditionalOptions>/noattributename</AdditionalOptions>
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:150]: Thanks for taking this on. I thought we were going to go with the other approach of not even producing the doc string info outside of the context of CI / producing the winmd api docs. I think that's cleaner than using an undocumented MIDL flag, mind changing it?

Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:150]: This issue from what I understand seems to be related to general attributes rather the doc string attribute specifically.  I worry that if we special case to doc strings, we may see the same issue in other contexts.

Comment (asklar) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:150]: the only attributes we have are the ones we created for doc purposes. Are you worried about us creating more custom attributes in the future? It's possible we would at some point but I'm not super worried about it right now

Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:150]: Yes, it is something where an attribute could be added to us, or inside of a community module, and we can see a break with no signal (unless we keep old versions in our repo).

**Review by dannyvv — APPROVED** [review]

**Pull request merged**